mvlearn/decomposition/groupica.py

```diff
@@ -0,0 +1,276 @@
+
+    unmixing_ : array, shape (n_components, n_components)
+        The square unmixing matrix, linking the output of the group-pca
+        and the independent components.
+
+    components_ : array, shape (n_components, n_components)
+        The square mixing matrix
+
+    individual_components_ : list of array
+        Individual unmixing matrices estimated by least squares.
+        `individual_components_[i]` is an array of shape
+        (n_components, n_features) where n_features is the number of
+        features in the dataset `i`.
+
+    individual_mixing_ : list of array
+        Individual mixing matrices estimated by least squares.
+        `individual_components_[i]` is an array of shape
+        (n_features, n_components) where n_features is the number of
+        features in the dataset `i`.
+
+    n_components_ : int
+        The estimated number of components.
+
+    n_features_ : list of int
+        Number of features in each training dataset.
+
+    n_samples_ : int
+        Number of samples in the training data.
+
+    n_views_ : int
+        Number of views in the training data
+
+    See also
+    --------
+    GroupPCA
+    multiviewica
+
+    References
+    ----------
+    .. [#1groupica] Calhoun, Vince, et al. "A method for making group
+                    inferences from functional MRI data using independent
+                    component analysis."
+                    Human brain mapping 14.3 (2001): 140-151.
+    Examples
+    --------
+    >>> from mvlearn.datasets import load_UCImultifeature
+    >>> from mvlearn.decomposition import GroupICA
+    >>> Xs, _ = load_UCImultifeature()
+    >>> ica = GroupICA(n_components=3)
+    >>> Xs_transformed = ica.fit_transform(Xs)
+    >>> print(len(Xs_transformed))
+    6
+    >>> print(Xs_transformed[0].shape)
+    (2000, 3)
+    """
+
+    def __init__(
+        self,
+        n_components=None,
+        n_individual_components="auto",
+        multiple_outputs=True,
+        prewhiten=False,
+        solver="picard",
+        ica_kwargs={},
+        random_state=None,
+    ):
+        if solver not in ["picard", "fastica"]:
+            raise ValueError(
+                "Invalid solver, must be either `fastica` or `picard`"
+            )
+        self.n_components = n_components
+        self.n_individual_components = n_individual_components
+        self.multiple_outputs = multiple_outputs
+        self.prewhiten = prewhiten
+        self.solver = solver
+        self.ica_kwargs = ica_kwargs
+        self.random_state = random_state
+
+    def fit(self, Xs, y=None):
+        r"""Fit  to the data and transform the data.
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+             - Xs length: n_views
+             - Xs[i] shape: (n_samples, n_features_i)
+        y : array, shape (n_samples,), optional
+
+        Returns
+        -------
+        sources : array, shape (n_samples, n_components)
```

Comment:
The return of `fit` is typically `self` and `sources_` is left as an attribute that can be accessed. Why the deviation here?